routini/src/proxy.rs

```diff
@@ -6,36 +8,85 @@ use pingora::{
     proxy::{ProxyHttp, Session},
     upstreams::peer::Tracer,
 };
-use tracing::instrument;
 
-use crate::load_balancing::{
-    LoadBalancer,
-    strategy::{Strategy, fewest_connections::ConnectionsTracer},
+use crate::{
+    load_balancing::{
+        LoadBalancer,
+        strategy::{Adaptive, fewest_connections::ConnectionsTracer},
+    },
+    server_builder::RouteConfig,
 };
 
-pub const DEFAULT_MAX_ALGORITHM_ITERATIONS: usize = 256;
+type MaxIterations = usize;
 
-pub struct LB<S: Strategy> {
-    pub load_balancer: Arc<LoadBalancer<S>>,
+pub struct RouteValue {
+    pub lb: Arc<LoadBalancer<Adaptive>>,
+    pub max_iterations: MaxIterations,
+    pub route_config: RouteConfig,
+}
+
+#[derive(Clone)]
+pub struct Proxy {
+    routes: Arc<Router<RouteValue>>,
+}
+
+impl Proxy {
+    pub fn new(routes: Router<RouteValue>) -> Self {
+        Proxy {
+            routes: Arc::new(routes),
+        }
+    }
+
+    pub fn route(&self, path: &str) -> Result<(&RouteValue, Option<String>)> {
+        tracing::info!("route path: {}", path);
+        self.routes
+            .at(path)
+            .map_err(|e| {
+                Box::new(Error {
+                    cause: Some(Box::new(e)),
+                    context: Some(ImmutStr::Static("Failed to route path to backend")),
+                    esource: ErrorSource::Internal,
+                    etype: ErrorType::HTTPStatus(StatusCode::BAD_REQUEST.as_u16()),
+                    retry: RetryType::Decided(false),
+                })
+            })
+            .map(|m| {
+                let value = m.value;
+                let stripped_path = if value.route_config.strip_path_prefix {
```

Comment:
You can continue with the fluent syntax by leveraging the fact that bool implements `then`.  For example:
```rs
let stripped_path = value.route_config.strip_path_prefix.then(|| m.params.... );
```
It's largely a matter of style and whether or not you feel it makes the code more clear and concise.